### PR TITLE
[dnb] Add new extractor

### PR DESCRIPTION
**Pull request by user706**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

New extractor for e.g. https://portal.dnb.de/audioplayer/do/show/1077188552

### CODE REVIEW

**Comment by user706** [comment]
strange...
Download only works correctly the 2nd time.

Example:
```bash
$ python3 -m youtube_dl https://portal.dnb.de/audioplayer/do/show/107679887X
[DNB] 107679887X: Downloading webpage
[download] Downloading playlist: None
[DNB] playlist None: Collected 1 video ids (downloading 1 of them)
[download] Downloading video 1 of 1
ERROR: unable to download video data: HTTP Error 404: Not Found

$ python3 -m youtube_dl https://portal.dnb.de/audioplayer/do/show/107679887X
[DNB] 107679887X: Downloading webpage
[download] Downloading playlist: None
[DNB] playlist None: Collected 1 video ids (downloading 1 of them)
[download] Downloading video 1 of 1
[download] Destination: Gavotte und Bourée aus der 'Französischen Suite Nr. 5' [Elektronische Ressource] _ Johann Sebastian Bach-107679887X.mp3
[download] 100% of 4.48MiB in 00:01
[download] Finished downloading playlist: None
```

File: youtube_dl/extractor/dnb.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+import re
+import json
+
+
+class DNBIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:portal\.dnb\.de/audioplayer/do/show/|d-nb\.info/)(?P<id>\w+)[/&]?'
```
Comment (dstftw) [inline youtube_dl/extractor/dnb.py]: `[/&]?` does not make any sense at the end.

File: youtube_dl/extractor/dnb.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            'title': 'Leonoren-Ouvertüre III [Elektronische Ressource] : Allegro / Ludwig van Beethoven (1770-1827)',
+            'id': '1077188552',
+            'ext': 'mp3',
+            'track': 'Track 1',
+            'track_number': 1
+        }
+    }]
+
+    @staticmethod
+    def update_and_return_dic(info_dict, update_info):
```
Comment (dstftw) [inline youtube_dl/extractor/dnb.py]: Inline.

File: youtube_dl/extractor/dnb.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        for obj in objs:
+            thumbnail = obj.get('cover_url')
+            if thumbnail:
+                thumbnail = 'https://portal.dnb.de/' + thumbnail
+
+            info_dict = {
+                'id': obj.get('idn'),
+                'title': obj.get('title'),
+                'author': obj.get('author'),
+                'url': 'https://portal.dnb.de/' + obj.get('media_url'),
```
Comment (dstftw) [inline youtube_dl/extractor/dnb.py]: `urljoin`.

File: youtube_dl/extractor/dnb.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+        m = re.search(r'fdnbpl.media\s*=\s*(\[.*\]);', webpage)
+        objs = json.loads(m.group(1))
+
+        result = []
+        num = 1
+        for obj in objs:
+            thumbnail = obj.get('cover_url')
+            if thumbnail:
+                thumbnail = 'https://portal.dnb.de/' + thumbnail
```
Comment (dstftw) [inline youtube_dl/extractor/dnb.py]: `urljoin`.

File: youtube_dl/extractor/dnb.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+            info_dict = {
+                'id': obj.get('idn'),
+                'title': obj.get('title'),
```
Comment (dstftw) [inline youtube_dl/extractor/dnb.py]: Mandatory. Read coding conventions.

File: youtube_dl/extractor/dnb.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        ret.update(update_info)
+        return ret
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        url = 'https://portal.dnb.de/audioplayer/do/show/' + video_id
+        webpage = self._download_webpage(url, video_id)
+
+        m = re.search(r'fdnbpl.media\s*=\s*(\[.*\]);', webpage)
+        objs = json.loads(m.group(1))
```
Comment (dstftw) [inline youtube_dl/extractor/dnb.py]: `_search_regex`, `_parse_json`. Again: read coding conventions.

**Pull request closed without merging**